google/cloud/bigquery/v2/minimal/internal/common_v2_resources.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigquery/v2/minimal/internal/common_v2_resources.h"
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+// Bypass clang-tidy warnings for self-referential and recursive structures.
+
+// NOLINTBEGIN
+struct QueryParameterType;
```

Comment:
Is this needed?

Reply:
Yeah..the forward declaration was needed because I was getting an error in the from_json() to to_json() method for QueryParameterStructType, that it couldn't find the QueryParameterType

NOLINTBEGIN/NOLINEND are needed to bypass the clang-tidy warning  due to circular and recursive references

Reply:
You probably do not need the forward deck in the .cc file.  It gets it from the .h (where it was needed originally)

Reply:
IIRC, the header file was needed to avoid the struct definition error and .cc was needed for json initialization via from_json and to_json.I can double check in the next PR and remove it if not needed.